src/main/java/net/neoforged/neoforge/codec/NBTSerializableCodec.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.codec;
+
+import static net.neoforged.neoforge.common.util.NeoForgeExtraCodecs.TAG_CODEC;
+
+import com.mojang.datafixers.util.Pair;
+import com.mojang.serialization.Codec;
+import com.mojang.serialization.DataResult;
+import com.mojang.serialization.DynamicOps;
+import java.util.function.Supplier;
+import net.minecraft.nbt.Tag;
+import net.minecraft.resources.RegistryOps;
+import net.neoforged.neoforge.common.util.INBTSerializable;
+
+/**
+ * A codec for instances of {@link INBTSerializable}.
```

Comment:
I think that this should be more specific, maybe something like
```suggestion
 * A codec for instances of {@link INBTSerializable} that can be constructed without reference to some game context.
```

Just mentioning that this is not suitable for NBT serializable types that require a reference to some context.